tfsdk/serve_import.go

```diff
@@ -32,7 +32,9 @@ func (r importedResource) toTfprotov6(ctx context.Context) (*tfprotov6.ImportedR
 			"Error converting imported resource response",
 			"An unexpected error was encountered when converting the imported resource response to a usable type. This is always a problem with the provider. Please give the following information to the provider developer:\n\n"+err.Error(),
 		)
-		return nil, diags
+		// Do not return a nil ImportedResource, it can cause Terraform CLI
+		// to panic.
+		return irProto6, diags
```

Comment:
Rather than returning empty objects, do we want to continue to return nil, and just handle the error here: https://github.com/hashicorp/terraform-plugin-framework/pull/164/files#diff-84c019e51c26505ea6d36a5b0012e60992a6dea901d5dd0fad7e3c2f961b3968R58

?

Reply:
This isn't empty, it'd include the passed through `TypeName`

Reply:
I'm not sure what it'd do on the other side though, since it is then calling:

```go
		state, err := decodeDynamicValue(imported.State, resSchema.Block.ImpliedType())
		if err != nil {
			resp.Diagnostics = resp.Diagnostics.Append(err)
			return resp
		}
```

So maybe it is best to omit the `ImportedResponse` altogether.